frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightContent.tsx

```diff
@@ -10,16 +10,19 @@ import {
 } from '@patternfly/react-core';
 import PipelineRunDrawerRightTabs from '~/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs';
 import { TaskReferenceMap, PipelineRunTaskDetails } from '~/concepts/pipelines/content/types';
+import { PipelineRunTaskParam } from '~/k8sTypes';
 
 type PipelineRunDrawerRightContentProps = {
   task?: PipelineRunTaskDetails;
   taskReferences: TaskReferenceMap;
+  parameter?: PipelineRunTaskParam[];
```

Comment:
Two things:
- This is an array of parameters so it would be better to use the plural "parameters" as the name of the prop.
- I think this prop should be required, not optional. Otherwise you end up having to make it optional all the way down the component tree and deal with the undefined case at the bottom of the call stack. IMO it would be better to check and pass in an empty array to this component. You then don't have to check if the array is defined further down the stack, you can just call an array methods (e.g. find, filter, map etc) knowing it will be a noop if there are no values.

Reply:
Ahh , keeping the parameter mandatory might  cause an error , because , as checked many pipeline don't have parameter

Reply:
@pnaik1 I understand, that's why I suggested checking to see if it is defined and passing an empty array in if it is undefined. 

That being said I don't think this is a big issue, especially now you've simplified the logic at the bottom of the component tree. I think this is fine as is :)